samples/snippets/create_fileset.py

```diff
@@ -0,0 +1,144 @@
+    # of a service account key file.
+    # -------------------------------
+    datacatalog = datacatalog_v1.DataCatalogClient()
+
+    # -------------------------------
+    # 1. Environment cleanup: delete pre-existing data.
+    # -------------------------------
+    # Delete any pre-existing Entry with the same name
+    # that will be used in step 3.
+    expected_entry_name = datacatalog_v1.DataCatalogClient.entry_path(
+        project_id, location, fileset_entry_group_id, fileset_entry_id
+    )
+
+    try:
+        datacatalog.delete_entry(name=expected_entry_name)
+    except (NotFound, PermissionDenied):
+        pass
+
+    # Delete any pre-existing Entry Group with the same name
+    # that will be used in step 2.
+    expected_entry_group_name = datacatalog_v1.DataCatalogClient.entry_group_path(
+        project_id, location, fileset_entry_group_id
+    )
+
+    try:
+        datacatalog.delete_entry_group(name=expected_entry_group_name)
+    except (NotFound, PermissionDenied):
+        pass
+
+    # -------------------------------
+    # 2. Create an Entry Group.
+    # -------------------------------
+    entry_group_obj = datacatalog_v1.types.EntryGroup()
+    entry_group_obj.display_name = "My Fileset Entry Group"
+    entry_group_obj.description = "This Entry Group consists of ...."
+
+    entry_group = datacatalog.create_entry_group(
+        parent=datacatalog_v1.DataCatalogClient.common_location_path(
+            project_id, location
+        ),
+        entry_group_id=fileset_entry_group_id,
+        entry_group=entry_group_obj,
+    )
+    print("Created entry group: {}".format(entry_group.name))
+
+    # -------------------------------
+    # 3. Create a Fileset Entry.
+    # -------------------------------
+    entry = datacatalog_v1.types.Entry()
+    entry.display_name = "My Fileset"
+    entry.description = "This fileset consists of ...."
+    entry.gcs_fileset_spec.file_patterns.append("gs://my_bucket/*")
+    entry.type_ = datacatalog_v1.EntryType.FILESET
+
+    # Create the Schema, for example when you have a csv file.
+    columns = []
+    columns.append(
+        datacatalog_v1.types.ColumnSchema(
+            column="first_name",
+            description="First name",
+            mode="REQUIRED",
+            type_="STRING",
+        )
+    )
+
+    columns.append(
+        datacatalog_v1.types.ColumnSchema(
+            column="last_name", description="Last name", mode="REQUIRED", type_="STRING"
+        )
+    )
+
+    # Create sub columns for the addresses parent column
+    subcolumns = []
+
+    subcolumns.append(
+        datacatalog_v1.types.ColumnSchema(
+            column="city", description="City", mode="NULLABLE", type_="STRING"
+        )
+    )
+
+    subcolumns.append(
+        datacatalog_v1.types.ColumnSchema(
+            column="state", description="State", mode="NULLABLE", type_="STRING"
+        )
+    )
+
+    columns.append(
+        datacatalog_v1.types.ColumnSchema(
```

Comment:
Might be more natural to create this column first and then append to `addresses_column.subcolumns`

https://proto-plus-python.readthedocs.io/en/stable/fields.html#repeated-fields